File: test/unit/code_parser/whole_tests/views/ocr_2017_rest.vb
' Copyright (c) 2000 ActivePDF, Inc.
' ActivePDF OCR 2017
' Example generated 01/01/2000 

Imports System
Imports System.IO
Imports System.Net
Imports System.Net.Http
Imports System.Net.Http.Headers

Public Class Examples
  Sub Example()
    Dim Folder As String = AppDomain.CurrentDomain.BaseDirectory
    Dim InputFile As String = Folder + "multipage.tif"
    Dim OutputFile As String = Folder + "new.pdf"
    Dim result As String = String.Empty
    
    Using client As New HttpClient()
        Dim token As String = String.Empty
        Dim uri As New Uri("http://localhost:62625/api/OCR/Conversion")
    
        ' Set the Access Token for authorization
        token = "<<insert token here>>"
        Dim ahv As New AuthenticationHeaderValue("Bearer", token)
        client.DefaultRequestHeaders.Authorization = ahv
    
        ' Multipart form post
        Dim content As New MultipartFormDataContent()
    
        ' Input file as a file stream
        Dim fs As FileStream = File.OpenRead(InputFile)
        Dim streamContent As New StreamContent(fs)
        streamContent.Headers.Add("Content-Type", "application/octet-stream")
        streamContent.Headers.Add("Content-Disposition", "form-data; name=""InputFile""; filename=""" + Path.GetFileName(InputFile) + """")
        content.Add(streamContent, "InputFile", Path.GetFileName(InputFile))
    
        ' Each setting can be passed in here as a new StringContent
        content.Add(New StringContent("0"), "OCRType")
        content.Add(New StringContent("1"), "PictureHandling")
        content.Add(New StringContent("4"), "PDFVersion")
        content.Add(New StringContent("1"), "Linearize")
    
        ' Post the content to the endpoint (uri)
        Dim hrm As HttpResponseMessage = client.PostAsync(uri, content).Result
        If hrm.StatusCode = HttpStatusCode.OK Then
            ' Get the resulting output of the post which is the PDF file created with OCR
            Dim bytes As Byte() = hrm.Content.ReadAsByteArrayAsync().Result
    
            ' Specify where to save the PDF
            File.WriteAllBytes(OutputFile, bytes)
        Else
            ' Get the error message if something went wrong
            Dim errorMsg As String = hrm.Content.ReadAsStringAsync().Result
            WriteResults(errorMsg)
        End If
    End Using
    ' Process Complete
    WriteResults("Done!")
  End Sub
  
  ' Write output data
  Sub WriteResults(content As String)
    ' Choose where to write out results
  
    ' Debug output
    'System.Diagnostics.Debug.WriteLine("ActivePDF: * " + content)
  
    ' Console
    Console.WriteLine(content)
  
    ' Log file
    'Using tw = New System.IO.StreamWriter(AppDomain.CurrentDomain.BaseDirectory & "application.log", True)
    '   tw.WriteLine("[" + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss") + "]: => " + content)
    'End Using
  End Sub
End Class

File: test/unit/code_parser/whole_tests/views/ocr_2017_rest.cs
// Copyright (c) 2000 ActivePDF, Inc.
// ActivePDF OCR 2017
// Example generated 01/01/2000 

using System;
using System.IO;
using System.Net;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Threading.Tasks;

class Examples
{
  public static void Example()
  {
    Task.Run(async () =>
    {
        string Folder = AppDomain.CurrentDomain.BaseDirectory;
        string Result = string.Empty;
        string InputFile = Folder + "multipage.tif";
        string OutputFile = Folder + "new.pdf";
    
        using (HttpClient client = new HttpClient())
        {
            string token = string.Empty;
    
            Uri uri = new Uri("http://localhost:62625/api/OCR/Conversion");
    
            // Set the Access Token for authorization
            token = "<<insert token here>>";
            AuthenticationHeaderValue ahv = new AuthenticationHeaderValue("Bearer", token);
            client.DefaultRequestHeaders.Authorization = ahv;
    
            // Multipart form post
            MultipartFormDataContent content = new MultipartFormDataContent();
    
            // Input file as a file stream
            FileStream fs = File.OpenRead(InputFile);
            StreamContent streamContent = new StreamContent(fs);
            streamContent.Headers.Add("Content-Type", "application/octet-stream");
            streamContent.Headers.Add("Content-Disposition", @"form-data; name=""InputFile""; filename=""" + Path.GetFileName(InputFile) + @"""");
            content.Add(streamContent, InputFile, Path.GetFileName(InputFile));
    
            // Each setting can be passed in here as a new StringContent
            content.Add(new StringContent("0"), "OCRType");
            content.Add(new StringContent("1"), "PictureHandling");
            content.Add(new StringContent("4"), "PDFVersion");
            content.Add(new StringContent("1"), "Linearize");
    
            // Post the content to the endpoint (uri)
            HttpResponseMessage hrm = await client.PostAsync(uri, content);
            if (hrm.StatusCode == HttpStatusCode.OK)
            {
                // Get the resulting output of the post
                byte[] bytes = await hrm.Content.ReadAsByteArrayAsync();
                // Specify where you want the PDF to save
                File.WriteAllBytes(OutputFile, bytes);
            }
            else
            {
                // Get the error message if something went wrong
                string error = await hrm.Content.ReadAsStringAsync();
                WriteResults(error);
            }
        }
    }).Wait();
    // Process Complete
    WriteResults("Done!");
  }
  
  // Write output data
  public static void WriteResults(string content)
  {
    // Choose where to write out results
  
    // Debug output
    //System.Diagnostics.Debug.WriteLine("ActivePDF: * " + content);
  
    // Console
    Console.WriteLine(content);
  
    // Log file
    //using (System.IO.TextWriter writer = new System.IO.StreamWriter(System.AppDomain.CurrentDomain.BaseDirectory + "application.log", true))
    //{
    //    writer.WriteLine("[" + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss") + "]: => " + content);
    //}
  }
}